### ISSUE TITLE
No Advice for Overrepresentation versus Enrichment Analysis

### ISSUE THREAD

**Issue by DarioS** [YuLab-SMU/biomedical-knowledge-mining-book#34]
Some students are using overrepresentation analysis even if only ten genes are D.E. There should be recommendations in which scenario to do overrepresentation (number of D.E. genes is large) and when to do enrichment (number of D.E. genes is small or 0).

**Comment by GuangchuangYu** [comment]
Good point — this was missing. Added a "Choosing between ORA and GSEA" section to the enrichment-algorithms chapter (e2cf0a4).

It makes the choice depend mainly on how many genes pass your threshold: many DEGs (dozens and up) → ORA; few (under roughly 20–30) or none at all → GSEA, because the hypergeometric test has almost no power on a short list and only a very strong overlap can reach significance.

It also spells out the failure mode you describe: running ORA on a handful of genes usually returns nothing, and that null result is easily misread as "no biology here" rather than "not enough data for this test".
